docs/rfc/rfc-6.adoc

```diff
@@ -0,0 +1,228 @@
+we do not want to block the release on this functionality. Maintainers will be
+reimbursed in ETH and may receive an additional reward for performing their
+duties.
+
+Transactions belonging to the third type can be executed by anyone and all
+off-chain clients will be responsible for submitting them. No ETH will be
+reimbursed and T coming from slashed stakes will be the only optional reward.
+No submission order will be maintained and all clients will try to report
+misbehavior at the same time. It is more important to keep the network healthy
+and free from misbehavior than to optimize costs.
+
+==== Group member incentives
+
+[[submit-dkg-result]]
+===== Submit + approve DKG result
+
+No submission order will be enforced on-chain and the transaction submitter will
+have ETH cost reimbursed. To disincentivize front-running, no additional reward
+will be released by this transaction. Off-chain clients should have a
+gentleman's agreement to follow the submission order based on
+`hash(new_group_pubkey) % group_size`.
+The submission order will be unenforceable but everyone will defaultly
+configured to follow the order anyway.
+
+The first member responsible for submitting the DKG result is a member with
+index `hash(new_group_pubkey) % group_size`. If that member did not submit the
+result,  `(hash(new_group_pubkey) % group_size) + 1` becomes responsible next.
+Group members who were inactive and skipped their responsibility might be
+reported by the wallet as ones who failed the heartbeat and marked as ineligible
+for rewards.
+
+For example, if `hash(new_group_pubkey) % group_size = 62`, `group_size = 64`,
+and member `9` submitted DKG result, members with indexes
+`{62, 63, 64, 1, 2, 3, 4, 5, 6, 7, 8}` might be marked as ineligible for
+rewards. Inactive group members are determined once the result is submitted and
+accepted by the chain.
+
+Given that the DKG result submission process consists of two transactions:
+submitting the result and approving the result after the challenge period ends,
+in order for the operator to be considered active, they must submit **both**
+transactions.
+
+===== Report heartbeat failed
+
+A wallet should periodically execute off-chain heartbeats and report inactive
+operators responsible for potential heartbeat failures. The majority of wallet
+members need to come together and sign a message with their private ECDSA
+operator keys. The message should contain the list of inactive operators and
+nonce ensuring uniqueness. The same mechanism can be used for reporting
+operators who failed their duty of submitting DKG result. 
+
+No submission order will be enforced on-chain for the heartbeat failure
+transaction and the transaction submitter will have ETH cost reimbursed. To
+disincentivize front-running, no additional reward will be released by this
+transaction. Off-chain clients should have a handshake agreement to follow the
+submission order: the operator submitting this TX is the first active one.
```

Comment:
> Or do we want to have a similar mechanism as we had for the result submission (hash(new_group_pubkey) % group_size)?

No, because this is about heartbeat failure, we are not in the context of DKG here. See https://github.com/keep-network/tbtc-v2/blob/main/docs/rfc/rfc-1.adoc#heartbeat.

Say we have a group of 5 operators: A,B,C,D,E and A,B are inactive and heartbeat failed because of them. The first to submit would be C, then D, then E but I intentionally left it a bit unspecified - we need to figure it out during the implementation, there are some more questions to answer here. For example, what happens if we have one operator inactive in the middle? One option would be to have C always responsible for submitting and if they don't, group signs another heartbeat failure with C included as inactive.